src/daemon/rpmostreed-transaction-types.c

```diff
@@ -678,12 +701,102 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
     }
   else if (self->override_reset_pkgs)
     {
-      if (!rpmostree_origin_remove_overrides (origin, self->override_reset_pkgs, error))
+      /* The origin stores removal overrides as pkgnames and replacement overrides as nevra.
+       * To be nice, we support both name & nevra and do the translation here by just
+       * looking at the commit metadata. */
+      OstreeDeployment *merge_deployment =
+        rpmostree_sysroot_upgrader_get_merge_deployment (upgrader);
+
+      gboolean is_layered;
+      g_autoptr(GVariant) removed = NULL;
+      g_autoptr(GVariant) replaced = NULL;
+      if (!rpmostree_deployment_get_layered_info (repo, merge_deployment, &is_layered, NULL,
+                                                  NULL, &removed, &replaced, error))
         return FALSE;
 
+      if (!is_layered)
+        return glnx_throw (error, "No overrides currently applied");
+
+      /* the strings are only stored once; in nevra_to_name */
```

Comment:
Is there a way to do this efficiently? Ideally, we'd just borrow the strings from the `removed` and `replaced` GVariants, but the strings are embedded inside a nested GVariant in there, and AFAICT, there doesn't seem to be a way to borrow directly from the parent GVariant rather than the one returned by `g_variant_get_child()`.